apps/application-system/api/src/app/modules/payment/payment.controller.ts

```diff
@@ -59,18 +61,39 @@ export class PaymentController {
   async createCharge(
     @CurrentUser() user: User,
     @Param('application_id', new ParseUUIDPipe()) applicationId: string,
+    @Body() payload: CreateChargeInput,
   ): Promise<CreatePaymentResponseDto> {
     if (!isUuid(applicationId)) {
       throw new BadRequestException(`ApplicationId is on wrong format.`)
     }
+    const allCatalogs = await this.paymentAPI.getCatalogByPerformingOrg(
+      '6509142520',
+    )
+
+    const catalog = await this.paymentService
+      .searchCorrectCatalog(
+        payload.chargeItemCode,
+        JSON.stringify(allCatalogs.item),
+      )
+      .catch((error) => {
+        throw new BadRequestException(
+          'Catalog request failed or bad input ' + error,
+        )
+      })
 
     const paymentDto: Pick<
       BasePayment,
-      'application_id' | 'fulfilled' | 'amount' | 'expires_at'
+      'application_id' | 'fulfilled' | 'amount' | 'definition' | 'expires_at'
     > = {
       application_id: applicationId,
       fulfilled: false,
-      amount: 8000,
+      amount: catalog.priceAmount,
+      definition: {
+        chargeItemName: catalog.chargeItemName,
+        chargeItemCode: catalog.chargeItemCode,
+        performingOrganiationID: catalog.performingOrgID,
+        chargeType: catalog.chargeType,
+      } as any,
```

Comment:
unsure what else I could use instead of any. Currently the definition is listed like this in the model
`  @Column({
    type: DataType.JSONB,
    defaultValue: {},
  })
  @ApiPropertyOptional()
  definition?: string`

Reply:
I see you're parsing this in another place

```typescript
const parsedDefinition = JSON.parse(payment.definition as string)
```

Shouldn't this then be stringified in the `DTO` and then you can lose the `any`?

```typescript
definition: JSON.stringify({
  chargeItemName: catalog.chargeItemName,
  chargeItemCode: catalog.chargeItemCode,
  performingOrganiationID: catalog.performingOrgID,
  chargeType: catalog.chargeType,
  amount: catalog.priceAmount,
})
```

Reply:
I changed model definition which brought on some other changes.